lib/realtime/vehicle.ex

```diff
@@ -132,24 +117,28 @@ defmodule Realtime.Vehicle do
     }
   end
 
-  @spec fraction_until_timepoint([StopTime.t()], Stop.id()) ::
-          {float(), StopTime.t() | nil} | nil
-  def fraction_until_timepoint([], _stop_id), do: {0.0, nil}
+  @spec timepoint_status([StopTime.t()], Stop.id()) :: timepoint_status() | nil
+  def timepoint_status([], _stop_id) do
```

Comment:
Is this extra case necessary? Will `Enum.split_while` and `Enum.find` just handle empty Lists and give you this result without an extra function definition?